django/db/models/functions/window.py

```diff
@@ -0,0 +1,119 @@
+from django.db.models import FloatField, Func, IntegerField
+
+__all__ = [
+    'CumeDist', 'DenseRank', 'FirstValue', 'Lag', 'LastValue', 'Lead',
+    'NthValue', 'Ntile', 'PercentRank', 'Rank', 'RowNumber',
+]
+
+
+class WindowFunction(Func):
+    window_compatible = True
+
+
+class CumeDist(WindowFunction):
+    function = 'CUME_DIST'
+    name = 'CumeDist'
+
+    def __init__(self, *expressions, **extra):
+        super().__init__(*expressions, output_field=FloatField())
```

Comment:
This is a recurring thing occurring here. What do you think about giving `Expression` a `get_output_field()` method and a `output_field` attribute that it can look up, @jarshwah ?